Input.cpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include "Engine.h"
+using namespace sf;
+
+
+void Engine::input()
+{
+	Event event;
+	updateMousePosition();
+
+	//Handle the player quitting
+	if (Keyboard::isKeyPressed(Keyboard::Escape))
+	{
+		m_Window.close();
+
+	}
+
+	////Handle the player starting the game
+	//if (Keyboard::isKeyPressed(Keyboard::Return))
+	//{m_Playing = true;} else {m_Playing = false;}
+
+	//Handle Mouse clicking
+	if (Mouse::isButtonPressed(Mouse::Left) && m_Playing == true)
+	{
+		
+		std::vector<sf::Sprite> &cupcakes = cupcake.m_spriteVector;
+		
+		for (int c = 0; c < cupcakes.size(); c++)
+		{
+			//if the cupcake has left the screen
+			if (cupcakes[c].getPosition().y >= m_Window.getSize().y)

```

Comment:
How does this work together with `Cupcake::removeCupcake()` function? That function also erases the cupcakes, but it does not remove the players health`.